### PR TITLE
chore: Upgrade to server v7 (PSD-37)

### PR DESCRIPTION
**Pull request by ShrutiTalewad**
[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

📓 &nbsp; **Related JIRA**

[PSD-37-0](https://genesisglobal.atlassian.net/browse/PSD-37)

🤔 &nbsp; **What does this PR do?**

Upgrade gradle for V7 of the framework

🚀 &nbsp; **Where should the reviewer start?**

Add file / directory pointers.

📑 &nbsp; **How should this be tested?**

```
npx -y @genesislcap/genx@latest init prtest -x --ref YOUR-BRANCH-NAME
```

✅ &nbsp; **Checklist**

- [ ] I have tested my changes.
- [ ] I have added tests for my changes.
- [ ] I have updated the project documentation to reflect my changes.

### CODE REVIEW

File: server/jvm/build.gradle.kts [hunk 1]
```diff
@@ -14,10 +14,10 @@ subprojects  {
     dependencies {
         implementation(platform("global.genesis:genesis-bom:${properties["genesisVersion"]}"))
         implementation("org.agrona:agrona:1.10.0!!")
```
Comment (kievitsp) [inline server/jvm/build.gradle.kts:16]: should probably get rid of this as well

**Review by kievitsp — COMMENTED** [review]
you'll also need to update the gradle jar and shell script

File: server/jvm/build.gradle.kts (hunk 1 above)
Comment (ShrutiTalewad) [reply server/jvm/build.gradle.kts:16]: Yes, I have noted that and similar we need to do for position-app-seed

File: server/jvm/build.gradle.kts [hunk 2]
```diff
@@ -13,24 +13,19 @@ subprojects  {
 
     dependencies {
         implementation(platform("global.genesis:genesis-bom:${properties["genesisVersion"]}"))
-        implementation("org.agrona:agrona:1.10.0!!")
-        testImplementation("org.jetbrains.kotlin:kotlin-test-junit:1.7.10")
+        testImplementation("org.jetbrains.kotlin:kotlin-test-junit:1.9.0")
```
Comment (jacinpoz) [inline server/jvm/build.gradle.kts:16]: should we just rely on the version provided by the kotlin plugin? 1.9.0 is different than 1.9.10. You should also be able to use the special kotlin dependency syntax:
`testImplementation(kotlin("test-junit"))`

File: server/jvm/{{appName}}-distribution/build.gradle.kts [hunk 3]
```diff
@@ -69,6 +69,10 @@ tasks {
         archiveBaseName.set("genesisproduct-{{appName}}")
         archiveClassifier.set("bin")
         archiveExtension.set("zip")
+        inputs.files(rootProject.getTasksByName("copyDependencies", true))
+    }
+    distTar {
+        inputs.files(rootProject.getTasksByName("copyDependencies", true))
```
Comment (jacinpoz) [inline server/jvm/{{appName}}-distribution/build.gradle.kts:75]: do we care about distTar at all? we only distribute zip files afaik

Comment (khouari1) [reply server/jvm/{{appName}}-distribution/build.gradle.kts]: I think, with the enhanced strictness in latest gradle version, it complains about it. If we don't care about the task then can we disable it?

Comment (khouari1) [reply server/jvm/{{appName}}-distribution/build.gradle.kts]: Disabled it.

File: .genx/tests/fixtures/testapp/docker-compose.yml [hunk 4]
```diff
@@ -4,16 +4,12 @@ services:
     tty: true
     container_name: gsf
     build:
-      context: server/jvm/
+      context: server/jvm/testapp-deploy/build/docker/
```
Comment (jacinpoz) [inline .genx/tests/fixtures/testapp/docker-compose.yml:7]: does "testapp" need to be templated?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -46,12 +46,16 @@ jobs:
...
-        run: ./gradlew assemble -PclientSpecific=true
+        run: ./gradlew build -PclientSpecific=true
 
       - name: Skip binding to privileged port on CI
         working-directory: /tmp/testapp
         run: sed -i "s/- '443:443'/# - '443:443'/g" docker-compose.yml
 
+      - name: Create Server Dockerfile
+        working-directory: /tmp/testapp
+        run: ./gradlew :genesisproduct-testapp:testapp-deploy:createDockerfile
```
Comment (jacinpoz) [inline .github/workflows/build.yml:57]: does "testapp" need to be templated?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -46,12 +46,16 @@ jobs:
 
       - name: Build sample app
         working-directory: /tmp/testapp
-        run: ./gradlew assemble -PclientSpecific=true
+        run: ./gradlew build -PclientSpecific=true
 
       - name: Skip binding to privileged port on CI
         working-directory: /tmp/testapp
```
Comment (jacinpoz) [inline .github/workflows/build.yml:52]: does "testapp" need to be templated?

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -46,12 +46,16 @@ jobs:
 
       - name: Build sample app
         working-directory: /tmp/testapp
```
Comment (jacinpoz) [inline .github/workflows/build.yml:48]: does "testapp" need to be templated?

File: .genx/tests/fixtures/testapp/docker-compose.yml (hunk 4 above)
Comment (khouari1) [reply .genx/tests/fixtures/testapp/docker-compose.yml:7]: No, this is just for the github build

File: .github/workflows/build.yml (hunk 5 above)
Comment (khouari1) [reply .github/workflows/build.yml:57]: No, this is just for the github build. Creates a new app, hardcoded to testapp, and builds it

File: .github/workflows/build.yml (hunk 7 above)
Comment (khouari1) [reply .github/workflows/build.yml:48]: Same as above

File: .github/workflows/build.yml (hunk 5 above)
Comment (khouari1) [reply .github/workflows/build.yml:57]: Looks like the .github folder is excluded from generated projects so is hardcoded here for testing

**Review by jacinpoz — APPROVED** [review]

**Pull request merged**